packages/node/src/indexer/fetch.service.ts

```diff
@@ -435,14 +441,18 @@ export class FetchService implements OnApplicationShutdown {
           this.eventEmitter.emit(IndexerEvent.SkipDictionary);
         }
       }
-      // the original method: fill next batch size of blocks
       const endHeight = this.nextEndBlockHeight(
         startBlockHeight,
         scaledBatchSize,
       );
-      this.blockDispatcher.enqueueBlocks(
-        range(startBlockHeight, endHeight + 1),
-      );
+
+      if (this.getModulos().length > 0) {
```

Comment:
I think this length needs to be equal to the number of datasources. If theres a block handler without modulo then we would only fetch modulo blocks.